plugins/transformations/sequence2array/srcCxx/Sequence2ArrayTransformation.cxx

```diff
@@ -245,7 +223,7 @@ bool Seq2ArrayTransformation::are_type_compatible(
         auto output_array_type = static_cast<const ArrayType &>(output_type);
 
         if (output_array_type.dimension_count() != 1) {
-            std::string error("Error: cannot transform a sequence to a "
+            std::string error("cannot transform a sequence to a "
                     "multidimensional array. The array dimension count should be 1.");
             throw std::runtime_error(error);
         }
```

Comment:
We can allow a sequence to be converted into a multi-dimensional array as long as we define a mapping between the input index and the "physical layout" of the array. The memory layout of multidimensional arrays is actually defined by XTypes (7.4.3.5.1, Table 40):
> Multi-dimensional arrays are treated for serialization purposes
> as a single dimensional array containing all the elements
> ordered such that the index of the first dimension varies most
> slowly, and the index of the last dimension varies most
> quickly.

Reply:
The conversion can be performed as long as we define a bijective mapping between an index `i` on the input sequence and a tuple `(i1, ..., iN)` of indices on the output array

Reply:
we can add that in a future version, right now this just supports sequence to 1D array